smart-contracts/assembly/contracts/NFT/NFT-internals.ts

```diff
@@ -1,282 +1,239 @@
-// ====                 APPROVAL                   ==== //
-// ==================================================== //
-
 /**
- * Store the approved address for a token
+ * Count all NFTs assigned to an owner.
  *
- * @param owner - owner address
- * @param spenderAddress - spender address
- * @param tokenId - The token ID to approve
+ * @param owner - An address for whom to query the balance
  */
-export function _approve(
-  tokenId: u64,
-  owner: string,
-  spenderAddress: string,
-): void {
-  assertIsMinted(tokenId);
-  assertIsOwner(owner, tokenId);
-  assert(!_isApproved(spenderAddress, tokenId), 'Already approved');
-
-  const key = approvedTokenKey + tokenId.toString();
-  Storage.set(key, spenderAddress);
+export function _balanceOf(owner: string): u256 {
+  const key = balanceKey(owner);
+  if (Storage.has(key)) {
+    return bytesToU256(Storage.get(key));
+  }
+  return u256.Zero;
 }
 
 /**
- * Removes the approval of the token
- * @param tokenId - the tokenID
+ * Returns the owner of a given tokenId.
+ *
+ * @param tokenId - The identifier for an NFT
+ * @returns the address of the owner of the NFT
  */
-function _removeApproval(tokenId: u64): void {
-  const key = approvedTokenKey + tokenId.toString();
-  Storage.set(key, '');
+export function _ownerOf(tokenId: u256): string {
+  const key = ownerKey(tokenId);
+  if (Storage.has(key)) {
+    return bytesToString(Storage.get(key));
+  }
+  return '';
 }
 
-export function _getApproved(tokenId: u64): string {
-  const key = approvedTokenKey + tokenId.toString();
-  if (!Storage.has(key)) return '';
-
-  return Storage.get(key);
+/**
+ * Returns the name of the contract.
+ */
+export function _name(): string {
+  if (Storage.has(NAME_KEY)) {
+    return bytesToString(Storage.get(NAME_KEY));
+  }
+  return '';
 }
 
-export function _isApproved(address: string, tokenId: u64): bool {
-  if (address.length === 0) return false;
-  const approvedAddress = _getApproved(tokenId);
-  return approvedAddress === address;
+/**
+ * Returns the symbol of the contract.
+ */
+export function _symbol(): string {
+  if (Storage.has(SYMBOL_KEY)) {
+    return bytesToString(Storage.get(SYMBOL_KEY));
+  }
+  return '';
 }
 
-export function _setApprovalForAll(
-  owner: string,
-  operator: string,
-  approved: bool,
-): void {
-  assert(owner != operator, 'Not allowed to set approval for all for yourself');
-
-  const key = approvedForAllTokenKey + owner + operator;
-
-  Storage.set(key, approved.toString());
+/**
+ * Change or reaffirm the approved address for an NFT
+ *
+ * @param tokenId - The NFT to approve
+ * @param approved - The new approved NFT controller
+ */
+export function _approve(approved: string, tokenId: u256): void {
+  assert(_isAuthorized(Context.caller().toString(), tokenId), 'Unauthorized');
+  const key = allowanceKey(tokenId);
+  if (approved == '') {
+    Storage.del(key);
+  } else {
+    Storage.set(key, stringToBytes(approved));
```

Comment:
You're not checking approved format. Is it on purpose ?

Reply:
yes, validation is to be done by the wrapper